pkg/expr/sql/db_test.go

```diff
@@ -205,6 +206,80 @@ func TestErrorsFromGoMySQLServerAreFlagged(t *testing.T) {
 	require.Contains(t, err.Error(), "error in go-mysql-server")
 }
 
+func TestFrameToSQLAndBack_JSONRoundtrip(t *testing.T) {
+	expectedFrame := &data.Frame{
+		RefID: "json_test",
+		Name:  "json_test",
+		Fields: []*data.Field{
+			data.NewField("id", nil, []*int64{p(int64(1)), p(int64(2))}),
+			data.NewField("payload", nil, []*json.RawMessage{
+				p(json.RawMessage(`{"foo":1}`)),
+				p(json.RawMessage(`{"bar":"baz"}`)),
+			}),
+		},
+	}
+
+	sqlDB := &DB{}
+
+	query := `SELECT * FROM json_test`
+
+	resultFrame, err := sqlDB.QueryFrames(context.Background(), "json_test", query, data.Frames{expectedFrame})
+	require.NoError(t, err)
+
+	// Ensure consistent names for comparison
+	resultFrame.Name = expectedFrame.Name
+	resultFrame.RefID = expectedFrame.RefID
+
+	if diff := cmp.Diff(expectedFrame, resultFrame, data.FrameTestCompareOptions()...); diff != "" {
+		require.FailNowf(t, "Frame mismatch (-want +got):\n%s", diff)
+	}
+}
+
+func TestQueryFrames_JSONFilter(t *testing.T) {
+	input := &data.Frame{
+		RefID: "A",
+		Name:  "A",
+		Fields: []*data.Field{
+			data.NewField("title", nil, []*string{
+				p("Bug report"),
+				p("Feature request"),
+			}),
+			data.NewField("labels", nil, []*json.RawMessage{
+				p(json.RawMessage(`["type/bug", "priority/high"]`)),
+				p(json.RawMessage(`["type/feature", "priority/low"]`)),
+			}),
```

Comment:
Again, none of these are null. Would this work?

```suggestion
			data.NewField("title", nil, []string{"Bug report", "Feature request"}),
			data.NewField("labels", nil, []json.RawMessage{
				json.RawMessage(`["type/bug", "priority/high"]`),
				json.RawMessage(`["type/feature", "priority/low"]`),
			}),
```

`expected` will probably need to change to match too